internal/exec/util/unit.go

```diff
@@ -110,6 +110,51 @@ func (ut Util) MaskUnit(unit types.Unit) (string, error) {
 	return filepath.Join("/", SystemdUnitsPath(), unit.Name), nil
 }
 
+// UnmaskUnit deletes the symlink to /dev/null for a masked unit
```

Comment:
It actually deletes any inode that's already there; we should either add a check or update the function name and docstring.

Reply:
You're right, but let me try to explain a bit.

In units.go I call `IsUnitMasked()` first and then only call `UnmaskUnit()` after it's checked. I did it this way instead of calling `IsUnitMasked()` from within `UnmaskUnit()` because I didn't want to litter the logs with a bunch of `unmasking unit ...` messages for things that weren't actually masked, which is undesirable.

One alternative here is to just add an extra call to `IsUnitMasked()` within `UnmaskUnit()` as well.

Reply:
Yup, that makes sense.  I'd just like to make the function a little safer, whether by documenting it more clearly or by adding the extra check.

Reply:
added another call to `IsUnitMasked()` in the latest upload